Docs: SceneApp (#172)
grafana/scenes: docusaurus/docs/scene-app-tabs.md

---
id: scene-app-tabs

> title: Tabs navigation to scene app

Reviewer: ```suggestion
title: Tab navigation in scene apps
```

---

`SceneAppPage` provides support for building Grafana applications with tabs navigation. Tabs navigation is especially useful for designing information architecture for complex Grafana apps, where visualizations are groupped into meaningful sections.

## Add tabs navigation to scene app

Defining tabs navigation for apps using scenes is a matter of utilizing `SceneAppPage` property `tabs`.

> ### Step 1. Create SceneApp

Reviewer: What we do here is dependent on what we decide [here](https://github.com/grafana/scenes/pull/172/files#diff-58a9a7146db52bf6ca49ddb0370dd504719a6dcccdc61e8c7c626a2e1a898a60R18)

Reviewer: ```suggestion
### Step 1. Create a Scenes app
```

> Follow [Building apps with scenes guide](./scene-app.md) to build your app.

Reviewer: ```suggestion
Follow the [Building apps with scenes guide](./scene-app.md) to build your app.
```

### Step 2. Create scenes for individual tabs

> Each tab renders its own scene, similar to `SceneAppPage`.

Reviewer: ```suggestion
Each tab renders its own scene, similar to `SceneAppPage`:
```

    },
    queries: [
      {
        refId: 'A',
        expr: 'sort_desc(avg by(handler) (rate(prometheus_http_request_duration_seconds_sum {}[5m]) * 1e3))',
        format: 'table',
        instant: true,
      },
    ],
  });

    return new EmbeddedScene({
    $data: queryRunner,
    body: new SceneFlexLayout({
      direction: 'column',
      children: [new SceneFlexItem({
        minHeight: 300,
        body: new VizPanel({
          title: 'HTTP Requests per handler',
          pluginId: 'table',
        })
      })],
    }),
  });
}
```

### Step 3. Configure tabs for page

Tabs are instances of `SceneAppPage` objects. Similarily to creating scene page, you create tabs. To render tabs, use `tabs` property of the `SceneAppPage` object.

```tsx

const overviewTab = new SceneAppPage({
  title: 'Overview',
  url: '/a/<PLUGIN_ID>/my-app/overview',
  getScene: getOverviewScene,
});

const handlersTab = new SceneAppPage({
  title: 'Handlers',
  url: '/a/<PLUGIN_ID>/my-app/handlers',
  getScene: getHandlersScene,
});

const myAppPage = new SceneAppPage({
  title: 'Grafana Scenes App',
  url: '`/a/<PLUGIN_ID>/my-app`,
  tabs: [
    overviewTab,
    handlersTab
  ]
});
```

> Navigating to `https://your-grafana.url/a/<PLUGIN_ID>/my-app` will render a scene app with two tabs: Overview and Handlers. Overview tab contains Time series panel with Prometheus HTTP Requests summary. Handlers tab contains table panel with Prometheus HTTP request average durration summary per handler.

Reviewer: ```suggestion
Navigating to `https://your-grafana.url/a/<PLUGIN_ID>/my-app` will render a scene app with two tabs: **Overview** and **Handlers**. The **Overview** tab contains a time series panel with a Prometheus HTTP Requests summary. The **Handlers** tab contains a table panel with a Prometheus HTTP request average duration summary, per handler.
```
